### DISCUSSION TITLE
Disk usage not updated in kubernetes, metrics fails

### DISCUSSION THREAD

**Discussion by schoenfbe** [minio/minio#15580] [Q&A]
I have a weird problem where MinIO doesn´t show used diskspace, and Monitoring/Metrics fails with 404, saying it has a license problem.
We're using a distributed setup with 4 drives, and if we upload anything, it shows up in MinIO and versioning/locking also works, but when we look at the disk usage in kubernetes, it tells us the usage is always 0%, no matter what we upload.
Generating files manually on each of the drives solves the problem temporarily, as metrics will then work, and kubernetes also reflects the correct usage.
However, new files uploaded will again not show up in usage, Metrics however keeps working.
Any ideas what could cause this, and how to resolve the issue?

Current MinIO version: 2022-05-08T23:50:31Z (tried that cause our normal system works with that)
RELEASE.2022-08-13T21-54-44Z and RELEASE.2022-08-22T23-53-06Z show the same issue

Thanks in advance :)

**Comment by klauspost** [comment]
@schoenfbe Could you post `mc admin info --json ALIAS` output?

**Reply by schoenfbe** [reply]
yes of course, but the error seems weird, as we're using an API port, at least thatś what we've configured.

`sh-4.4# mc admin info --json minio-dist
{
    "status": "error",
    "error": "S3 API Request made to Console port. S3 Requests should be sent to API port.",
    "info": {
        "buckets": {
            "count": 0
        },
        "objects": {
            "count": 0
        },
        "versions": {
            "count": 0
        },
        "usage": {
            "size": 0
        },
        "services": {
            "kms": {},
            "ldap": {}
        }
    }
}`

**Reply by schoenfbe** [reply]
but with that output, to me it kinda looks like that might be the problem, I just don't know how to fix it.

**Reply by schoenfbe** [reply]
If it helps, I can also post the helm command used to generate the yaml, maybe the issue is there, since we've never used a distributed setup before

**Reply by schoenfbe** [reply]
@klauspost another thing I should mention, when we look into kubernetes directly and look at the disk usage with kubectl, it's also shows 0% usage until we generate files via the command line.

**Reply by klauspost** [reply]
@schoenfbe You have mc configured to use the wrong port. You should configure it with the API port, not the console port.

**Reply by schoenfbe** [reply]
@klauspost thanks :)
maybe stupid question: how do I do that in helm or yaml?

the current helm command looks like this:

`helm template minio-for-dvc-dev-distributed --namespace namespacename --set namespace=namespacename --set persistence.size=1Gi --set persistence.storageClass=drivename --set replicas=1 --set resources.requests.memory=1Gi --set resources.limits.cpu=1000m --set resources.limits.memory=1Gi --set mode=distributed --set ingress.enabled=true --set ingress.hosts={minio-dvc-dev-distributed.....} --set consoleIngress.enabled=true --set drivesPerNode=4 --set consoleIngress.hosts={minio-dvc-dev-distributed-console....} --set environment.MINIO_BROWSER_REDIRECT_URL=https://minio-dvc-dev-distributed-console...../ --set DeploymentUpdate.maxUnavailable=100% --set DeploymentUpdate.maxSurge=0 --set ingress.annotations.nginx\\.ingress\\.kubernetes\\.io\\/proxy-body-size=500m --set users= minio/minio`

I just edited out the URLs and the names

**Reply by schoenfbe** [reply]
oh right, and the used repo is helm repo add minio https://charts.min.io/

**Reply by klauspost** [reply]
```
                        "totalspace": 1073741824,
                        "availspace": 1073741824,
```                        
Seems like your disks are not returning usable values. We forward those values.

Look into your filesystem space reporting.

**Reply by schoenfbe** [reply]
@klauspost thanks for confirming, we also got to suspecting that, we'll look into it.
Might be that our system doesn't handle this properly with the distributed setup, since it works fine with standalone.
But at least we know where to look :)

**Reply by schoenfbe** [reply]
@klauspost follow up question, how would usable values look like? or just simply that those two numbers should be different, so the available space really represents the difference to total space, as an integer value?
thanks

**Reply by klauspost** [reply]
It should just be the remaining space and total available space of the disk in bytes.
